### PR TITLE
[5.0] Fix, Hide toolbar in modal when it not in use

### LINKED ISSUE #41572: [5.0] Toolbar present in ModalSelect dialog
### Steps to reproduce the issue

compare 4.4 or 5.0-alpha 4 with latest 5.0-dev (contains #40462) @Fedik 

### Expected result

no toolbar present

### Actual result

toolbar present, eg
image
image

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #41572 .

### Summary of Changes
Hide toolbar on empty state, in modal.
Because `$this->setLayout('emptystate');` override the layout value the condition `if ($this->getLayout() !== 'modal')` never work.

### Testing Instructions
On installation without articles.
Try select an article for menu. Also Category, Contact, Tags

Also try to add a module to dashboard.

### Actual result BEFORE applying this Pull Request
You get a toolbar in modal

### Expected result AFTER applying this Pull Request
No toolbar

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

Reference:
- #40462

### CODE REVIEW

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 63b034b0459cdf7e80763609abb2ca97060eee5b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41578">issues.joomla.org/tracker/joomla-cms/41578</a>.</sub>

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 63b034b0459cdf7e80763609abb2ca97060eee5b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41578">issues.joomla.org/tracker/joomla-cms/41578</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41578">issues.joomla.org/tracker/joomla-cms/41578</a>.</sub>

**Comment by Fedik** [comment]
Should be good now also for Media.
Can only re-test "image select" while aricle editing

**Comment by richard67** [comment]
Back to pending.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41578">issues.joomla.org/tracker/joomla-cms/41578</a>.</sub>

**Comment by obuisard** [comment]
Fedir @Fedik the media modal is now working properly

**Comment by obuisard** [comment]
Got 3 different results while testing after an update, trying now on new install.

image

**Comment by Fedik** [comment]
> Got 3 different results while testing after an update

:white_check_mark:  "New article" uses new ModalField, so it is okay ,
:white_check_mark:  "New contact" uses old ModalField, so it also okay ,
:x: "New newsfeed" uses old ModalField, and it need to fix the toolbar,

**Comment by obuisard** [comment]
> > Got 3 different results while testing after an update
> 
> ✅ "New article" uses new ModalField, so it is okay , ✅ "New contact" uses old ModalField, so it also okay , ❌ "New newsfeed" uses old ModalField, and it need to fix the toolbar,

I confirm. Also
:white_check_mark: "New categories"

Could not find where tags could be created in modal.

**Comment by Fedik** [comment]
newsfeed should be good now

File: administrator/components/com_media/tmpl/file/default.php [hunk 1]
```diff
@@ -36,13 +36,6 @@
 
 $input->set('hidemainmenu', true);
 
-// Load the toolbar when we are in an iframe
-if ($tmpl == 'component') {
-    echo '<div class="subhead noshadow">';
-    echo Toolbar::getInstance('toolbar')->render();
```
Comment (Quy) [inline administrator/components/com_media/tmpl/file/default.php:42]: Please remove import statement.

Comment (Fedik) [reply administrator/components/com_media/tmpl/file/default.php:42]: done

File: administrator/components/com_modules/src/View/Select/HtmlView.php [hunk 2]
```diff
@@ -67,7 +67,10 @@ public function display($tpl = null)
             throw new GenericDataException(implode("\n", $errors), 500);
         }
 
-        $this->addToolbar();
+        if ($this->getLayout() !== 'modal') {
```
Comment (Quy) [inline administrator/components/com_modules/src/View/Select/HtmlView.php:70]: Sorry to nitpick. In the other files, you assigned it to `$isModal`.

Comment (Quy) [inline administrator/components/com_modules/src/View/Select/HtmlView.php:70]: Sorry to nitpick. In the other files, you assigned it to `$isModal`.

Comment (Fedik) [reply administrator/components/com_modules/src/View/Select/HtmlView.php:70]: it need there, because `emptystate` layout, and here it not in use

**Comment by Fedik** [comment]
I will set it as Draft for now. 
There other related issues, wich need to find a better approach.

**Comment by Fedik** [comment]
Please test alternative fix #41600

**Pull request closed without merging**